catalyst_voices/packages/catalyst_voices_repositories/lib/src/credentials_storage_repository.dart

```diff
@@ -9,20 +9,22 @@ import 'package:result_type/result_type.dart';
 /// It will be replaced by a proper implementation as soon as authentication
 /// flow will be defined.
 final class CredentialsStorageRepository {
-  final SecureStorageService secureStorageService;
+  final Storage _storage;
 
-  const CredentialsStorageRepository({required this.secureStorageService});
+  const CredentialsStorageRepository({
+    required Storage storage,
+  }) : _storage = storage;
 
-  void get clearSessionData => secureStorageService.deleteAll;
+  Future<void> get clearSessionData async => _storage.clear();
 
   Future<Result<SessionData?, SecureStorageError>> getSessionData() async {
     try {
-      final email = await secureStorageService.get(
-        SecureStorageKeysConst.dummyEmail,
+      final email = await _storage.readString(
```

Comment:
I like to have the middle layer that wraps storage calls, here it would be:
`String readEmail()`
`setEmail(String email)`

Why? Because it's easier to make mistakes when passing keys than having concrete signatures.

What do you reckon?

Reply:
Yup, i like idea. Ill create `DummyAuthStorage` which will extend `Storage`